src/tokens/saga.ts

```diff
@@ -233,25 +238,47 @@ interface UserBalancesResponse {
 export async function fetchTokenBalancesForAddress(
   address: string
 ): Promise<FetchedTokenBalance[]> {
-  const response = await apolloClient.query<UserBalancesResponse, { address: string }>({
-    query: gql`
-      query FetchUserBalances($address: Address!) {
-        userBalances(address: $address) {
-          balances {
-            tokenAddress
-            balance
+  const chainsToFetch = getFeatureGate(StatsigFeatureGates.FETCH_MULTI_CHAIN_BALANCES)
+    ? Object.values(networkConfig.networkToNetworkId)
+    : [networkConfig.defaultNetworkId]
+  const userBalances = await Promise.all(
+    chainsToFetch.map((networkId) => {
+      return apolloClient.query<UserBalancesResponse, { address: string; networkId: string }>({
+        query: gql`
+          query FetchUserBalances($address: Address!, $networkId: NetworkId) {
+            userBalances(address: $address, networkId: $networkId) {
+              balances {
+                tokenId
+                tokenAddress
+                balance
+              }
+            }
           }
-        }
-      }
-    `,
-    variables: {
-      address,
-    },
-    fetchPolicy: 'network-only',
-    errorPolicy: 'all',
-  })
-
-  return response.data.userBalances.balances
+        `,
+        variables: {
+          address,
+          networkId: networkId.replaceAll('-', '_'), // GraphQL does not support hyphens in enum values
+        },
+        fetchPolicy: 'network-only',
+        errorPolicy: 'all',
+      })
+    })
+  )
+  return userBalances.reduce(
+    (acc, response) => acc.concat(response.data.userBalances.balances),
+    [] as FetchedTokenBalance[]
+  )
```

Comment:
```suggestion
  return userBalances.flatMap(
    (response) => response.data.userBalances.balances
  )
```
code golf

Reply:
ohh sick -- wouldn't have thought to use `flatMap`!